include/SW_Site.h

```diff
@@ -230,12 +230,11 @@ void PTF_RawlsBrakensiek1985(
 RealD calculate_soilBulkDensity(RealD matricDensity, RealD fractionGravel);
 RealD calculate_soilMatricDensity(RealD bulkDensity, RealD fractionGravel,
 								  LOG_INFO* LogInfo);
-LyrIndex nlayers_bsevap(SW_LAYER_INFO** lyr, LyrIndex n_layers);
+LyrIndex nlayers_bsevap(RealD *evap_coeff, LyrIndex n_layers);
 void nlayers_vegroots(LyrIndex n_layers, LyrIndex n_transp_lyrs[],
-					  SW_LAYER_INFO** lyr);
+					  RealD transp_coeff[][MAX_LAYERS + 1]);

```

Comment:
Would it be possible to elaborate a bit more on what you mean by a new define? I understand the concern for the +1 but I don't think I understand what you're proposing.

Reply:
There are two questions
1. Why do we need these arrays to be `MAX_LAYERS + 1`?
2. If so, would something like `#define MAX_LAYERS_PLUS1 26 // MAX_LAYERS_PLUS1 = MAX_LAYERS + 1` be more efficient? Or does it not matter (much)?

Reply:
Thank you for the clarification, here are my answers:

1.  This is to make room for a deep drain layer, which when running severe tests with size `MAX_LAYERS` proved that it was not sufficient space (test `SWFlowTranspWeightedAvg`, or `TranspWeightedAvg` if it's before I push my changes), hence `MAX_LAYERS + 1`.

2. The only impact on efficiency (if any) from adding something like that would be is the compilation time since this memory would be allocated then. I believe your idea would be good from the standpoint of readability (and maybe maintainability). So if we want to be descriptive, it would be a good idea to make a new define for it, like you suggested.

Reply:
Ok, thanks, makes sense! Though I still not convinced that setting up the soil layer arrays for `MAX_LAYERS + 1` is the best approach -- I'll think about re-organizing deep drainage. Maybe create a dedicated attribute in "SW_SOILWAT" instead of stuffing it into `SW_SoilWat.swcBulk[][]`? Let's discuss

Reply:
Resolved with commit aec25159be5d42bcd9817cc103c062010693be16 "Deep drainage is now percolation from deepest soil layer" -> no need anymore for `MAX_LAYERS + 1`